include/deal.II/fe/mapping_q1.h

```diff
@@ -80,6 +80,10 @@ class MappingQ1 : public MappingQGeneric<dim, spacedim>
  * these contexts throughout the library, this class defines a static $Q_1$
  * mapping object. This object can then be used in all of those places where
  * such an object is needed.
+ *
+ * @note The use of this object should be avoided since it is only applicable
+ *   in cases where a mesh consists exclusively of quadrilaterals or hexahedra.
+ *   Use ReferenceCell::get_default_linear_mapping() instead.
```

Comment:
Is this a candidate for DEPRECATED_EARLY? Or is this too early? :-)

Reply:
Yes, somewhere in this series of patches.